### PR TITLE
feat: add time-series-decomposition modifier

### PR DESCRIPTION
**Pull request by Anuj0047**
Changes to add new modifier for time-series-decomposition line chart. 
This change is part of new time-series-decomposition analysis type feature in insights.

image

image

### CODE REVIEW

File: src/expression-modifiers/time-series-decomposition/__tests__/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+  beforeEach(() => {
+    modifier = {
+      type: 'timeSeriesDecomposition',
+      disabled: false,
+      decomposition: '',
+      steps: 1,
+      outputExpression: '',
+    };
+    expression = 'Sum(Sales)';
+    properties = {
```
Comment (veinfors) [inline src/expression-modifiers/time-series-decomposition/__tests__/index.spec.js:20]: I suggest you move this properties object into it's own file.

**Comment by veinfors** [comment]
Please add a description to the PR where you explain what this is and give some info. Otherwise it will be hard to review the PR. It's also good for tracability and knowledge sharing.

**Comment by Anuj0047** [comment]
> Please add a description to the PR where you explain what this is and give some info. Otherwise it will be hard to review the PR. It's also good for tracability and knowledge sharing.

@veinfors Updated PR description and made suggested change. Can please check? thanks!

File: src/index.js [hunk 2]
```diff
@@ -717,6 +719,10 @@ function applyMeasureModifiers({
         }
         availableModifiers[modifier.type].initModifier(modifier);
 
+        if (modifier.type === 'timeSeriesDecomposition' && modifier.decomposition === '') {
```
Comment (quanho) [inline src/index.js:722]: This logic should be moved to inside initModifier in the time-series-decomposition/index.js file

Comment (Anuj0047) [reply src/index.js]: Made  requested change, thanks for the review!

**Review by quanho — APPROVED** [review]

**Pull request merged**